src/static/js/modules/loan-comparison/components/styled-numeric-input.js

```diff
@@ -1,12 +1,20 @@
 var React = require('react');
 var common = require('../common');
 var TextInput = require('./input-text');
+var nonNumericRegex = /[^0-9.]+/g;
 
-// adds a unit (dollar or percent sign) to input
-var StyledTextInput = React.createClass({
+// Enforce numeric input
+var numericValidator = function (val) {
```

Comment:
We should discuss with designers how this should work post-user testing - I would prefer an error message if you try to type with text to avoid confusion